modules/elements-color-picker/assets/js/editor/commands/enter-preview.js

```diff
@@ -2,8 +2,22 @@ import CommandBase from 'elementor-api/modules/command-base';
 
 export class EnterPreview extends CommandBase {
 	apply( args ) {
+		const { container, kit } = this.component.currentPicker;
+
 		// Silent
-		this.component.currentPicker.container.settings.set( this.component.currentPicker.control, args.value );
-		this.component.currentPicker.container.view.renderUI();
+		container.settings.set( control, args.value );
+
+		const { view } = container;
+
+		if ( view?.renderUI ) {
+			view.renderUI();
+		}
+
+		if ( kit ) {
+			const { id } = kit.config,
+				cssVar = `--e-global-color-${ container.id }`;
+
+			elementor.$previewContents[ 0 ].querySelector( `.elementor-kit-${ id }` ).style.setProperty( cssVar, args.value );
+		}
```

Comment:
```suggestion
			const { id } = kit.config,
				cssVar = `--e-global-color-${ container.id }`;

			elementor.$previewContents[ 0 ].querySelector( `.elementor-kit-${ id }` ).style.setProperty( cssVar, args.value );
		}
```

you do it twice, you can have a function in component to handle it.

Reply:
Yup, kinda stupid that I haven't done that at the first time.